soupsieve/util.py

```diff
@@ -67,6 +67,27 @@ def uchr(i):
         return struct.pack('i', i).decode('utf-32')
 
 
+class SelectorSyntaxError(SyntaxError):
+    """Syntax error in a CSS selector."""
+
+    def __init__(self, msg, pattern, index):
+        """Initialize."""
+
+        self.line = pattern.count('\n', 0, index) + 1
+        self.col = index - pattern.rfind('\n', 0, index)
+        offset = 3
+        text = []
+        for n, line in enumerate(pattern.split('\n'), start=1):
+            text.append(('--> {}' if n == self.line else '    {}').format(line))
```

Comment:
Suggestion: skip the --> for the (I expect most common) case where pattern is a single-line string

Reply:
That's a good idea 👍 .